site/content/blog/2023-01-19-authentication-help/index.md

```diff
@@ -0,0 +1,118 @@
+
+### Turning Authentication Off
+
+You should always aim to [make your life easier](/docs/authentication/make-your-life-easier/).
+
+If you can completely disable authentication in a safe environment then that is an ideal option.
+
+Unfortunately this is not possible with many apps.
+
+### Why Not Handle Authentication Yourself?
+
+You may be able to handle authentication yourself without configuring ZAP to understand it.
+
+If your app allows access using a specific authentication header which you can generate on demand then you can tell ZAP to use that via [authentication environmental variables](/docs/authentication/handling-auth-yourself/#authentication-env-vars).
+
+You can also write scripts (either ZAP ones or external ones) which login to your app and then inject the relevant authentication tokens in the right places.
+
+However, you should be aware that handling authentication yourself is risky:
+
+1. ZAP will try to test everything it finds, unless you specifically configure it not to.
+Most apps will have an option to Log Out - if ZAP uses this then from that point on its requests may be unauthenticated.
+1. ZAP also submits lots of requests that look malicious. Some applications may detect these sort of requests and then log ZAP out.
+1. Apps change over time. If the authentication process changes then it may break your scripts and ZAP will once again be scanning unauthenticated.
+
+### Why Configure ZAP to Understand Authentication?
+
+ZAP has a more sophisticated understanding of authentication.
+
+If configured correctly then ZAP will be able to tell whether or not its requests are authenticated and if they are not then it will automatically authenticate again.
+
+It also maintains a set of [statistics](/internal-statistics/) which allow you to see whether or not ZAP is successfully authenticating to your app.
+The [Automation Framework](/docs/automate/automation-framework/) makes it easy to [test statistics](/docs/desktop/addons/automation-framework/authentication/#authentication-statistics) so that automated tests can fail if the authentication starts to fail.
+
+### What Does ZAP Need to Know?
+
+ZAP needs to know the following information:
+
+* How to make an authentication (login) request
+* How the app's session handling works
+* How to tell if requests from ZAP are authenticated (the verification strategy)
+
+ZAP may also need to know how to store authentication tokens in a browser, depending on how your client side code works.
+
+To be able to configure ZAP to handle authentication you therefore need to know _exactly_ how your app's authentication and session handling works as well as how your client side handles session tokens.
+
+This can be tricky to find out, and if you configure any of it slightly wrong then authentication will fail.
+
+### Help is on its Way!
+
+We know authentication is hard to figure out and configure in ZAP, but we have plans to make it easier.
```

Comment:
I was trying to word this part so that it didn’t seem unique to ZAP. All DAST have this challenge.

Reply:
good point

Reply:
"We know authentication is hard to figure out and configure in all security tools, but we have plans in to make it easier in ZAP."?

Reply:
"much easier"? 😀

Reply:
Great!

Reply:
Changed to that, but happy to tweak again..